Realm/ObjectServerTests/setup_baas.sh

```diff
@@ -0,0 +1,167 @@
+#!/bin/sh
+
+######################################
+#
+# Copyright 2020 Realm Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http:#www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+######################################
+
+mongodb_version=4.2.5
+go_version=1.13
+node_version=8.11.2
+stitch_rev=19bb7818c31e2e071f32d3caf9ac8a96916bb68e
+mongodb_url="https://fastdl.mongodb.org/osx/mongodb-macos-x86_64-$mongodb_version.tgz"
+transpiler_target="node8-macos"
+server_stitch_lib_url="https://s3.amazonaws.com/mciuploads/mongodb-mongo-master/stitch-support/macos-debug/e791a2ea966bb302ff180dd4538d87c078e74747/stitch-support-4.3.2-721-ge791a2e-patch-5e2a6ad2a4cf473ae2e67b09.tgz"
+
+BASE_DIR=`pwd`/Realm/ObjectServerTests/build
+PID_FILE="$BASE_DIR/pid.txt"
+
+function setup_mongod() {
+    set -e
+    if [ ! -d "mongodb-macos-x86_64-$mongodb_version" ]
+    then
+    curl --silent ${mongodb_url} | tar xz
+    pushd mongodb-*
+    mkdir db_files
+    popd
+    fi
+}
+
+function run_mongod() {
+    set -e
+    pushd mongodb-*
+    echo "starting mongod..."
+    ./bin/mongod --dbpath ./db_files --port 26000 --replSet test --pidfilepath $PID_FILE
+    popd
+}
+
+function wait_for_mongod() {
+    set -e
+    pushd mongodb-*
+    echo "waiting for mongod to start up"
+    ./bin/mongo --nodb --eval 'assert.soon(function(x){try{var d = new Mongo("localhost:26000"); return true}catch(e){return false}}, "timed out connecting")'
+    ./bin/mongo --port 26000 --eval 'rs.initiate()'
+    echo "mongod is up."
+    popd
+}
+
+function shutdown_mongod() {
+    set -e
+    pushd mongodb-*
+    ./bin/mongo --port 26000 admin --eval "db.adminCommand({replSetStepDown: 0, secondaryCatchUpPeriodSecs: 0, force: true})"
+    ./bin/mongo --port 26000 admin --eval "db.shutdownServer()"
+    echo "mongod is down."
+    popd
+}
+
+function setup_stitch() {
+    set -e
+    OG_DIR=`pwd`
+    echo "setting up stitch"
+    if [ ! -d stitch ]
+    then
+    echo "cloning stitch"
```

Comment:
The lack of indentation of blocks in this script make the control flow very hard to follow.